app/src/main/java/com/example/concordia_campus_guide/Helper/RoutesHelpers/DirectionsApiDataParser.java

```diff
@@ -51,6 +56,59 @@ protected DirectionsResult doInBackground(DirectionsApiDataRetrieval... obj) {
      */
     @Override
     protected void onPostExecute(DirectionsResult result) {
-        dataRetrieval.caller.directionsApiCallBack(result);
+        dataRetrieval.caller.directionsApiCallBack(result, extractRelevantInfoFromDirectionsResultObj(result));
+    }
+
+    public List<Route> extractRelevantInfoFromDirectionsResultObj(DirectionsResult result) {
+        List<Route> routeOptions = new ArrayList<>();
+        for(DirectionsRoute directionsRoute: result.routes) {
+            Route route = null;
+            if(dataRetrieval.transportType.equals(ClassConstants.DRIVING)) {
```

Comment:
why not a switch statement?

Reply:
honestly, I don't think adding a comment is always good. I think if you break down the method into smaller ones, each one with a great title that explains what is going, we won't need comment. it might seem a bit useless but it helps a ton when doing unit tests, the smaller a method, the easier it is to test

Reply:
http://nicolecarpenter.github.io/2016/03/08/clean-code-chapter-4-comments.html This is a summary from the comment chapter of clean code which I reference a lot in the PR comment.

Reply:
While I like comments in general, I think proper naming of methods and variables is the first step

Reply:
by comment i meant put java documentation but i agree that i should break down this method